pkg/virtualmachine/enricher/enricher_impl.go

```diff
@@ -1,65 +1,47 @@
 package enricher
 
 import (
-	"context"
-
-	"github.com/google/go-containerregistry/pkg/name"
 	"github.com/pkg/errors"
 	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
 	"github.com/stackrox/rox/generated/storage"
-	"github.com/stackrox/rox/pkg/env"
 	"github.com/stackrox/rox/pkg/logging"
-	"github.com/stackrox/rox/pkg/scanners/scannerv4"
-	"github.com/stackrox/rox/pkg/scannerv4/client"
+	"github.com/stackrox/rox/pkg/scanners/types"
 )
 
 var (
-	log         = logging.LoggerForModule()
-	scanTimeout = env.ScanTimeout.DurationSetting()
+	log = logging.LoggerForModule()
 )
 
-const vmMockDigest = "vm-registry/repository@sha256:900dc0ffee900dc0ffee900dc0ffee900dc0ffee900dc0ffee900dc0ffee900d"
-
 type enricherImpl struct {
-	scannerClient client.Scanner
+	vmScanner types.VirtualMachineScanner
 }
 
-func New(scannerClient client.Scanner) VirtualMachineEnricher {
+func New(scanner types.VirtualMachineScanner) VirtualMachineEnricher {
 	return &enricherImpl{
-		scannerClient: scannerClient,
+		vmScanner: scanner,
 	}
 }
 
 func (e *enricherImpl) EnrichVirtualMachineWithVulnerabilities(vm *storage.VirtualMachine, indexReport *v4.IndexReport) error {
 	// Clear any pre-existing notes
 	vm.Notes = vm.GetNotes()[:0]
 
-	if e.scannerClient == nil {
+	if e.vmScanner == nil {
 		vm.Notes = append(vm.Notes, storage.VirtualMachine_MISSING_SCAN_DATA)
 		return errors.New("Scanner V4 client not available for VM enrichment")
 	}
 
-	if indexReport == nil {
-		vm.Notes = append(vm.Notes, storage.VirtualMachine_MISSING_SCAN_DATA)
-		return errors.New("index report is required for VM scanning")
-	}
-
-	vmDigest, err := name.NewDigest(vmMockDigest)
-	if err != nil {
-		vm.Notes = append(vm.Notes, storage.VirtualMachine_MISSING_SCAN_DATA)
-		return errors.Wrapf(err, "failed to parse digest for VM %q", vm.GetName())
-	}
-
-	ctx, cancel := context.WithTimeout(context.Background(), scanTimeout)
-	defer cancel()
-
-	vr, err := e.scannerClient.GetVulnerabilities(ctx, vmDigest, indexReport.GetContents())
+	scan, err := e.vmScanner.GetVirtualMachineScan(vm, indexReport)
```

Comment:
The image and node enrichment workflows use a semaphore to not overwhelm the scanner, ideally these semaphores would be shared (just saw they were not which IMO is a bug, created ROX-32039 as a result)

Consider using the same mechanism to protect scanner from being overwhelmed.